nitro-node/src/scripts/download-nitro.ts

```diff
@@ -0,0 +1,220 @@
+import fs from "node:fs";
+import path from "node:path";
+import download from "download";
+import { Duplex } from "node:stream";
+import crossFetch from "cross-fetch";
+
+// Define nitro version to download in env variable ("latest" or tag "v1.2.3")
+const NITRO_VERSION = process.env.NITRO_VERSION || "latest";
+// The platform OS to download nitro for
+const PLATFORM = process.env.npm_config_platform || process.platform;
+// The platform architecture
+//const ARCH = process.env.npm_config_arch || process.arch;
+
+const getReleaseInfo = async (taggedVersion: string): Promise<any> => {
```

Comment:
I prefer to skip node-gyp at this stage. As I understand, the primary use-case of node-gyp is to build node native modules (eg. when we want to build nitro into *.so files and use node binding to control nitro directly instead of going through a subprocess wrapper).
Using node-gyp for just getting the binaries is somehow overkill, and supporting programmatic build on every supported platform is another story. At this point I'm not confident enough that I can support it without people raising issues (build env involves PATH, OS versions, available tooling, etc..). For now, I don't think stepping into it will make good use of the time, more issues will happen...

Reply:
The intention for now is simple, to ease the usage of nitro in Nodejs programs without too much boilerplate code repeated and bloated between projects. It's not about overhead but rather about practical usage, it's just a simple wrapper for now, with no intention to bring surplus value to what nitro already has but rather improve developer experience on making a client program that utilizes nitro. 😅

Reply:
For building nitro during installation if no binary is available for the platform, let's move it to another issue as I believe it will be rather complex. Let's deliver what is useful for gaining more adoption of nitro from fellow developers then improve it later with more features...😅

Reply:
FYI: I'm looking at a reference build script at the link below
https://github.com/withcatai/node-llama-cpp/blob/master/src/cli/commands/BuildCommand.ts

They also don't use node-gyp but rather have a build script themself. I also prefer it this way, to not rely on node-gyp if not necessary means no dependency on Python and other native libs required by node-gyp, then we can have a less demanding library => more usage from other projects => more visibility, and more contribution back to nitro and this library. 😉